javascript/javascript2/week3/main.js

```diff
@@ -0,0 +1,136 @@
+//1)Log out the text Called after 2.5 seconds 2.5 seconds after the script is loaded.
+setTimeout(() => console.log("Called after 2.5 seconds "), 2500);
+
+//2.Create a function that takes 2 parameters: delay and stringToLog
+function timerString(delay, stringToLog) {
+  return setTimeout(() => {
+    console.log(stringToLog);
+  }, delay * 1000);
+}
+timerString(7, "Hello 7");
+timerString(3, "Hello 3");
+timerString(1, "Hello 1");
+
+//3.Create a button in html. When clicking this button, use the function you created in the previous task to log out the text:
+// Called after 5 seconds 5 seconds after the button is click
+let myBtn = document.querySelector(".timed-text");
+console.log(myBtn);
+myBtn.addEventListener("click", (callback) => {
+  timerString(5, "5seconds ended");
+});
+
+//4.Create two functions and assign them to two different variables.
+
+earthLogger = function () {
+  console.log("You asked to logout Earth");
+};
+saturnLogger = function () {
+  console.log("You asked to logout Saturn");
+};
+
+function planedLogFunction(planet) {
+  planet();
+}
+planedLogFunction(earthLogger);
+planedLogFunction(saturnLogger);
+
+//5.Create a button with the text called "Log location".
+let locationBtn = document.createElement("button");
+locationBtn.className = "location-btn";
+locationBtn.innerHTML = "Log location";
+body = document.querySelector("body");
+body.appendChild(locationBtn);
+function success(geolocationPosition) {
+  console.log("Success");
+  console.log(geolocationPosition);
+}
+
+function error(error) {
+  console.log(error.message);
+}
+locationBtn.addEventListener("click", () => {
+  navigator.geolocation.getCurrentPosition(success, error);
+});
+
+//7.Create a function called runAfterDelay. It has two parameters: delay and callback
+function runAfterDelay(CallBack, delay) {
```

Comment:
CallBack with a capital C is not the best name for your function, prefer using cammelCase in JS :)